app/audit/AuditService.scala

```diff
@@ -31,19 +30,20 @@ import javax.inject.Inject
 import scala.concurrent.ExecutionContext
 import scala.xml.NodeSeq
 
-class AuditService @Inject()(auditConnector: AuditConnector, messageTranslator: MessageTranslation)(implicit ec: ExecutionContext) {
+class AuditService @Inject() (auditConnector: AuditConnector, messageTranslator: MessageTranslation)(implicit ec: ExecutionContext) {
 
   def authAudit(auditType: AuditType, details: AuthenticationDetails)(implicit hc: HeaderCarrier): Unit =
     auditConnector.sendExplicitAudit(auditType.toString, details)
 
-  def auditEvent(auditType: AuditType, customerId: Ior[TURN, EORINumber], message: Message, channel: ChannelType)(implicit hc: HeaderCarrier): Unit = {
-    val details = AuthenticatedAuditDetails(channel, customerId, messageTranslator.translate(toJson(message.message)))
+  def auditEvent(auditType: AuditType, customerId: String, enrolmentType: String, message: Message, channel: ChannelType)(implicit hc: HeaderCarrier): Unit = {
+    val details = AuthenticatedAuditDetails(channel, customerId, enrolmentType, messageTranslator.translate(toJson(message.message)))
```

Comment:
`auditEvent` should accept an `EnrolmentId` and this method should break it down for the `AuthenticatedAuditDetails` - something like:

```suggestion
  def auditEvent(auditType: AuditType, enrolmentId: EnrolmentId, message: Message, channel: ChannelType)(implicit hc: HeaderCarrier): Unit = {
    val details = AuthenticatedAuditDetails(channel, enrolmentId.customerId, enrolmentId.enrolmentType, messageTranslator.translate(toJson(message.message)))
```

That way the fact that we are using an `EnrolmentId` here is enforced, rather than two strings which isn't so much enforced.

Reply:
...oh